include/net/net_pkt.h

```diff
@@ -178,6 +178,12 @@ struct net_pkt {
 	u8_t ieee802154_rssi; /* Received Signal Strength Indication */
 	u8_t ieee802154_lqi;  /* Link Quality Indicator */
 #endif
+
+#if defined(CONFIG_NET_SOCKET_RAW)
+	u16_t l2_proto;   /* IEEE 802.3 protocol value of the L2 (in network
```

Comment:
Let's put it in a union, I guess above 116 to 140 could be the right place: there are 2 u8_t, which would need to be combined in an anonymous struct, and that u16_t. something like that

Also, can you call it raw_proto  instead?